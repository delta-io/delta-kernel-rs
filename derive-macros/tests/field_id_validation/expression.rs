use delta_kernel_derive::ToSchema;

const MY_ID: i64 = 123;

#[derive(ToSchema)]
struct TestStruct {
    #[field_id = MY_ID]  // Should be literal, not expression
    field: String,
}

fn main() {}
