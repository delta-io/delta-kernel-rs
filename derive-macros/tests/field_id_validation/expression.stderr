error: field_id error: Expected field-id to be an integer
 --> tests/field_id_validation/expression.rs:7:18
  |
7 |     #[field_id = MY_ID]  // Should be literal, not expression
  |                  ^^^^^

error: attribute value must be a literal
 --> tests/field_id_validation/expression.rs:7:18
  |
7 |     #[field_id = MY_ID]  // Should be literal, not expression
  |                  ^^^^^
